docs/source/metabase.py

```diff
@@ -0,0 +1,174 @@
+"""
+A Sphinx builder putting documentation on Metabase.
+
+"""
+
+import re
+from typing import Any, Dict, List
+
+import sphinx.writers.text
+import requests
+from docutils.core import publish_string
+from sphinx.application import Sphinx
+from sphinx.builders.dummy import DummyBuilder
+from sphinx.util import logging
+from sphinx.writers.text import TextTranslator, TextWriter
+
+logger = logging.getLogger(__name__)
+
+# Such a shame to depend on globals…
+sphinx.writers.text.MAXWIDTH = float('inf')
+
+
+class MetabaseBuilder(DummyBuilder):
+    """Sphinx builder used to deploy descriptions on Metabase."""
+    name = 'metabase'
+    default_translator_class = TextTranslator
+    tables = None
+    writer = None
+
+    def _get(self, path: str) -> Dict[str, Any]:
+        """Launch a GET requests on Metabase API."""
+        response = requests.get(f'{self.config.metabase_url}/{path}', headers=self.headers)
+        response.raise_for_status()
+        return response.json()
+
+    def _put(self, path: str, json: Dict[str, Any]) -> Dict[str, Any]:
+        """Launch a PUT requests on Metabase API."""
+        response = requests.put(
+            f'{self.config.metabase_url}/{path}', headers=self.headers, json=json)
+        response.raise_for_status()
+        return response.json()
+
+    @property
+    def headers(self) -> Dict[str, Any]:
+        """Get headers used for requests on Metabase API."""
+        return {
+            'Content-Type': 'application/json',
+            'X-Metabase-Session': self.config.metabase_session,
```

Comment:
Since the session token expires, I think you'll need to add some code to get one from `/api/session` using normal credentials. I also think you'll want to have a way to get the credentials from environment variables. It kinda seems like this is coming from a sphinx config file, but I'm not familiar with that interface.

Reply:
Of course, that would be much easier. We’ll store the login and the password as environment variables in the CI, and give them to the CLI. They’re hidden in the logs, at least with Travis.

Reply:
I would still suggest getting them out of environment variables in azafea instead of marshaling them in as CLI arguments. It's less secure and adds more failure cases to handle. It would probably ultimately be less code than marshaling them through the CLI. Not a showstopper, though.

Reply:
Authentication now uses username/password instead of session token. Properties can be given as environment variables or Sphinx properties (useful when developing, doesn’t require much more code).